File: designs/publication/views/tag.js
/* PubRefDb: Publication database web application.
   Index publication documents by tag.
   Value: null.
*/
function(doc) {
    if (doc.entitytype !== 'publication') return;
    if (!doc.tags) return;
    for (var i in doc.tags) {
	emit(doc.tags[i], null);
    }
}

File: tag.py
""" PubRefDb: Publication database web application.

List of publications having a given tag.
"""

from .base import *


class Tag(MethodMixin, GET):
    "List of publications having a given tag."

    outreprs = [JsonRepresentation,
                MedlineRepresentation,
                AtomRepresentation,
                PublicationsListHtmlRepresentation]

    def get_data_resource(self, request):
        tag = request.variables['tag']
        publications = self.get_docs('publication/tag', tag)
        self.sort_publications(publications)
        for publication in publications:
            self.normalize_publication(publication, request.application.get_url)
        return dict(title="Publications: %s" % tag,
                    resource='Publication list tag',
                    publications=publications)
